src/zeekygen/Target.cc

```diff
@@ -501,7 +501,7 @@ vector<string> dir_contents_recursive(string dir)
 	while ( dir[dir.size() - 1] == '/' )
 		dir.erase(dir.size() - 1, 1);
 
-	char* dir_copy = util::copy_string(dir.c_str());
+	char* dir_copy = util::copy_string(dir.c_str(), dir.size());
 	char** scan_path = new char*[2];
 	scan_path[0] = dir_copy;
```

Comment:
Doesn't matter which one you use, since `data()` and `c_str()` are *required* by the standard (since C++11) to be equivalent (https://en.cppreference.com/w/cpp/string/basic_string/data).

Reply:
What I meant is why do we need to make the copy when we could possibly do `scan_path[0] = dir.data()` ?

Reply:
`fts_open` seems to expect mutable pointers to the two character arrays. I don't know if any implementation will ever actually touch the path, but we need to produce two `char*` here. We could do something underhanded like `&dir[0]`, but I don't know if it'd be safe. I've never used `fts_open` and the documentation I could find isn't very explicit about whether or not the function is allowed to change `path_argv[0]`.